### DISCUSSION TITLE
Distance limit on Network Hubs?

### DISCUSSION THREAD

**Discussion by Tyrasuki** [fnuecke/oc2#211] [Q&A]
Hi all,

I'm trying to do a little experiment where I run a simulated ISP for a village, but it seems after about 5 or 6 network hubs chained together, a computer cannot reach the primary router, nor other connected clients.

I'm not sure if this has to do with physical distance, or perhaps with a internal limit of the amount of hubs that can be daisy-chained?

Manually adding an ARP entry for v4 doesn't seem to work, and the ipv6 ULU (fe80::/64) ranges seem to be unreachable as well.
I've also set the TTL for all ping attempts to 128. since the documentation mentioned one hub counts as one hop.

Any clues? :)

I've attached some screenshots of the hubs and their layout as well as the connected devices.

### TOPOLOGY AND RESULTS
![3oCLJMM 1](https://user-images.githubusercontent.com/12734805/175978522-be747010-9bbd-4234-85c7-bea8a05d6307.jpg)
![4FtYUKv 1](https://user-images.githubusercontent.com/12734805/175976787-7f34c406-3db3-406c-b787-403e2b96889c.png)
![8ACJw2g 1](https://user-images.githubusercontent.com/12734805/175977864-6d7f02db-f13a-436a-937b-f9aa2a1b553f.png)
![5wijpzT 1](https://user-images.githubusercontent.com/12734805/175979055-01a80299-414c-4c2d-b57f-47838aeed047.png)

**Comment by samuelh2005** [comment]
I'm also curious if there is a distance limit, becuase I'm geting similar problems with my ISP too.

I have a computer that is within a 40 block radius of my ISP router and it is not able to receive DHCP packets, but a computer that is next to the ISP router can.

**Reply by PatriikPlays** [reply]
The TTL seems to be decremented on connectors as well, I'm not sure if that is intended or not, but you can get your setup working by increasing `admin.virtual_network.ethernetFrameTimeToLive` in `config/oc2-common.toml` to account for this.
Assuming default TTL (12):
![image](https://github.com/user-attachments/assets/b95dba57-fc16-4b0f-bcc5-de9fa2521a2d)

**Reply by PatriikPlays** [reply]
Oh, connectors can have 2 connections and act like a hub. I see why they decrement the TTL now, but the default TTL seems low.

![image](https://github.com/user-attachments/assets/34b3ff2d-154a-4033-98cb-cefb8e24df4e)
![image](https://github.com/user-attachments/assets/f368c455-a294-4cdf-8571-176ad6e21ef2)

**Reply by samuelh2005** [reply]
Thanks, gald someone could help.

**Reply by CoolCat467** [reply]
So in the end in summary, the solution is to raise the time to live value of packets instead of leaving it at its default value

**Reply by samuelh2005** [reply]
> So in the end in summary, the solution is to raise the time to live value of packets instead of leaving it at its default value

Yes, since each network connector decreases the TTL, increasing the TTL will mean that the packet can travel further. So, using a TTL of 9999 (or more) should be enough for large networks.

**Reply by Tyrasuki** [reply]
Interesting, I wasn't aware the TTL was so low on default settings, that does make sense though.
Thanks for the brainstorming. :)

Will mark this as solved as this seems to work.

**Reply by CoolCat467** [reply]
Curious, what is the default time to live if you happen to know?

**Reply by samuelh2005** [reply]
> Curious, what is the default time to live if you happen to know?

I'm not sure what the default one is, but if you not changed the config file the value will be there. If you have chnaged it, you could start a tempoaray Minecraft server so the mod will generate a new config.
